### PR TITLE
Update Copy task to log full paths

### LINKED ISSUE #9141: [Bug]: The Copy task should log absolute paths
### Issue Description

Similar to https://github.com/dotnet/msbuild/pull/3506, but when hardlink is used.

Originally, #3506 was to fix https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/180, and it seems like https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/180 is still an issue when using hardlinks.

### Steps to Reproduce

Have a build with the following properties set:

```xml
		<CreateHardLinksForCopyFilesToOutputDirectoryIfPossible>true</CreateHardLinksForCopyFilesToOutputDirectoryIfPossible>

		<CreateHardLinksForCopyLocalIfPossible>true</CreateHardLinksForCopyLocalIfPossible>
		<CreateHardLinksForAdditionalFilesIfPossible>true</CreateHardLinksForAdditionalFilesIfPossible>
		<CreateHardLinksForPublishFilesIfPossible>true</CreateHardLinksForPublishFilesIfPossible>

```

### Expected Behavior

The Copy task should log full absolute path.

### Actual Behavior

The Copy task logs relative paths. The writes to `Project1/bin/Release/File.dll` and `Project2/bin/Release/File.dll` will both be logged as `bin/Release/File.dll` and will be considered as double writes even though they are really different. So basically the same issue as https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/180

### Analysis

_No response_

### Versions & Configurations

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #9141

### CODE REVIEW

File: src/Tasks/Copy.cs [hunk 1]
```diff
@@ -361,7 +361,7 @@ private void LogDiagnostic(string message, params object[] messageArgs)
         private void TryCopyViaLink(string linkComment, MessageImportance messageImportance, FileState sourceFileState, FileState destinationFileState, out bool linkCreated, ref string errorMessage, Func<string, string, string, bool> createLink)
         {
             // Do not log a fake command line as well, as it's superfluous, and also potentially expensive
-            Log.LogMessage(MessageImportance.Normal, linkComment, sourceFileState.Name, destinationFileState.Name);
+            Log.LogMessage(MessageImportance.Normal, linkComment, sourceFileState.FileNameFullPath, destinationFileState.FileNameFullPath);
```
Comment (JanKrivanek) [inline src/Tasks/Copy.cs:364]: Is `FileNameFullPath` allways guaranteed to be populated here?
Should we coallesce with `Path.GetFullPath(<fileState>.Name)` or at least `<fileState>.Name`?

**Review by JanKrivanek — APPROVED** [review]
Looks good to me - though I have one comment for consideration.

Generally it is very prefered to first wait for the item to go through the internal triage - to prevent wasted efforts in case team has different idea about the bug (not a high risk with oneliner like this - but just mentioning it to prevent any confusion in the future)

File: src/Tasks/Copy.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Tasks/Copy.cs:364]: I aligned this with the existing code: https://github.com/dotnet/msbuild/blob/0125fc9fb838951c8750135c71336aa1f5e868a9/src/Tasks/Copy.cs#L341

But to answer the question, yes it's always guaranteed. The full path is populated in this call:

https://github.com/dotnet/msbuild/blob/9ae833e5fc229f521fc017d0ece4b9353289d699/src/Tasks/Copy.cs#L756

The actual copy that will try to utilize the full path happens here: https://github.com/dotnet/msbuild/blob/9ae833e5fc229f521fc017d0ece4b9353289d699/src/Tasks/Copy.cs#L767

Comment (JanKrivanek) [reply src/Tasks/Copy.cs:364]: Perfect - thanks for confirming

**Pull request merged**